src/test/net_tests.cpp

```diff
@@ -246,6 +246,25 @@ BOOST_AUTO_TEST_CASE(cnetaddr_basic)
     BOOST_CHECK(!addr.IsBindAny());
     BOOST_CHECK_EQUAL(addr.ToString(), "1122:3344:5566:7788:9900:aabb:ccdd:eeff");
 
+    // IPv6, scoped/link-local. See https://tools.ietf.org/html/rfc4007
+    // We support non-negative decimal integers (uint32_t) as zone id indices.
+    // Test with a fairly-high value, e.g. 32, to avoid locally reserved ids.
+    const std::string link_local{"fe80::1"};
+    const std::string scoped_addr{link_local + "%32"};
+    BOOST_REQUIRE(LookupHost(scoped_addr, addr, false));
+    BOOST_REQUIRE(addr.IsValid());
+    BOOST_REQUIRE(addr.IsIPv6());
+    BOOST_CHECK(!addr.IsBindAny());
+    const std::string addr_str{addr.ToString()};
+    BOOST_CHECK(addr_str == scoped_addr || addr_str == "fe80:0:0:0:0:0:0:1");
+    // The fallback case "fe80:0:0:0:0:0:0:1" is needed for macOS 10.14/10.15 and (probably) later.
+    // Test that the delimiter "%" and default zone id of 0 can be omitted for the default scope.
```

Comment:
I saw this check fail in a recent cirrus run `[previous releases, uses qt5 dev package and some depends packages] [unsigned char] [bionic]`:

https://cirrus-ci.com/task/5950587750580224?logs=ci#L4679

I'm restarting the test, will see if happens again.

Reply:
Thanks for reporting. It may be hitting a reserved id; have had it happen on occasion locally and bumping the `32` to a higher value resolved it. What happened after restarting?
```cpp
    // Test with a fairly-high value, e.g. 32, to avoid locally reserved ids.
    const std::string link_local{"fe80::1"};
    const std::string scoped_addr{link_local + "%32"};
```

Reply:
Thanks, interesting! (I don't know what a reserved id is and googling didn't seem to turn up anything helpful, so I'd be curious for any pointers.)

But test passed on restart so it is an intermittent issue like you suggested. Maybe the test can be written so the check is less strict and just depends on parsing behavior not OS behavior. But it doesn't seem too urgent if this is a known infrequent issue.

Reply:
My bad for ambiguous wording! `s/reserved id/id in use/`...will fix issue and wording.